src/test/java/springboot/webapp/usersmanager/services_tests/UserServiceTests.java

```diff
@@ -0,0 +1,144 @@
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.test.web.servlet.*;
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import springboot.webapp.usersmanager.controllers.UserController;
+import springboot.webapp.usersmanager.entities.Role;
+import springboot.webapp.usersmanager.entities.User;
+import springboot.webapp.usersmanager.repositories.UserRepository;
+import springboot.webapp.usersmanager.services.UserService;
+
+import java.util.*;
+
+import static org.hamcrest.Matchers.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest(UserController.class)
+public class UserServiceTests {
+    @Autowired
+    MockMvc mockMvc;
+
+    @Autowired
+    ObjectMapper mapper;
+
+    @MockBean
+    UserRepository userRepository;
+
+    @MockBean
+    UserService userService;
+
+    User user1 = new User(1, "Sergey", "Gonchar", "sergey@gmail.com", Role.ROLE_ADMIN);
+
+    User user2 = new User(2, "Sergey", "Danilov", "danilov@gmail.com", Role.ROLE_USER);
+
+    User user3 = new User(3, "Ivan", "Petrov", "petrov@gmail.com", Role.ROLE_USER);
+
+    @Test
+    public void getAll_usersList_ok() throws Exception {
+        List<User> users = new ArrayList<>(Arrays.asList(user1, user2, user3));
+
+        when(userService.getAll()).thenReturn(users);
+
+        mockMvc.perform(MockMvcRequestBuilders
+                        .get("/users")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$", hasSize(3)));
+    }
+
+    @Test
+    public void getById_existedUser_ok() throws Exception {
+        when(userService.get(user1.getId())).thenReturn(Optional.of(user1));
+
+        mockMvc.perform(MockMvcRequestBuilders
+                        .get("/users/1")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$", notNullValue()))
+                .andExpect(jsonPath("$.name", is("Sergey")))
+                .andExpect(jsonPath("$.surname", is("Gonchar")));
+    }
+
+    @Test
+    public void getById_nonExistentUser_notFound() throws Exception {
+        when(userService.get(user1.getId())).thenReturn(Optional.of(user1));
+
+        mockMvc.perform(MockMvcRequestBuilders
+                        .get("/users/2")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isNotFound());
```

Comment:
getStatus and check in AssertThat
read about benefits
https://objectpartners.com/2013/09/18/the-benefits-of-using-assertthat-over-other-assert-methods-in-unit-tests/